Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp

```diff
@@ -1412,57 +1396,17 @@ AffineTransform GraphicsContextCG::getCTM(IncludeDeviceScale includeScale) const
     return CGContextGetCTM(platformContext());
 }
 
-std::optional<std::pair<float, float>> GraphicsContextCG::scaleForRoundingToDevicePixels() const
+FloatRect GraphicsContextCG::roundToDevicePixels(const FloatRect& rect) const
 {
-    if (m_userToDeviceTransformKnownToBeIdentity)
-        return std::nullopt;
-
-    CGAffineTransform deviceMatrix = CGContextGetUserSpaceToDeviceSpaceTransform(platformContext());
-    if (CGAffineTransformIsIdentity(deviceMatrix)) {
-        m_userToDeviceTransformKnownToBeIdentity = true;
-        return std::nullopt;
+    CGAffineTransform deviceMatrix;
+    if (!m_userToDeviceTransformKnownToBeIdentity) {
+        deviceMatrix = CGContextGetUserSpaceToDeviceSpaceTransform(contextForState());
+        if (CGAffineTransformIsIdentity(deviceMatrix))
+            m_userToDeviceTransformKnownToBeIdentity = true;
```

Comment:
I think `m_userToDeviceTransformKnownToBeIdentity` should be `std::optional<bool>` which is set to `std::nullopt` when the transform changes. And here it can be set to 

```
m_userToDeviceTransformIsIdentity = CGAffineTransformIsIdentity(deviceMatrix)
```

So we do not to have to call CGAffineTransformIsIdentity() every time if the transform is not identity.

Reply:
Ok, but should we do it in this patch or in some other patch?